### PR TITLE
Selector values list

### PR DESCRIPTION
**Pull request by collina**

### CODE REVIEW

File: documentation/examples/interconnection_study_example.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "file_format_version": 1.1,
-    "site": "lga02",
-    "client_provider": "comcast",
-    "start_time": "2014-02-01T00:00:00Z",
+    "site": ["lga01", "lga02"],
```
Comment (mtlynch) [inline documentation/examples/interconnection_study_example.json]: I think we should change the names to `sites`, `client_providers`, etc now that it's lists

File: documentation/examples/interconnection_study_example.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+    "client_provider": ["comcast", "verizon", "cablevision"],
+    "start_time": ["2014-02-01T00:00:00Z"],
     "duration": "30d",
-    "metric":"minimum_rtt",
+    "metric":"download_throughput",
```
Comment (mtlynch) [inline documentation/examples/interconnection_study_example.json]: Any reason not to make this a list?

File: main.py [hunk 3]
```diff
@@ -38,6 +38,9 @@
 import telescope.utils
 
 
+MAX_THREADS_NORMAL_MODE = 18
+MAX_THREADS_BATCH_MODE = 100
```
Comment (mtlynch) [inline main.py]: This is kind of weird. This shows up in both 5bf17f9 (expected) and bf48a14 (unexpected) even though the second one isn't a merge. Any idea what happened here?

File: main.py [hunk 4]
```diff
@@ -292,11 +295,11 @@ def generate_query(selector, ip_translator, mlab_site_resolver):
 
   server_ips = []
   try:
-    for retrieved_site_ip in mlab_site_resolver.get_site_ips(selector.site_name,
+    for retrieved_site_ip in mlab_site_resolver.get_site_ips(selector.site,
```
Comment (mtlynch) [inline main.py]: This is a good idea. I filed an issue on myself in Observatory to get rid of the `_name` suffix there too:

https://github.com/opentechinstitute/mlab-observatory/issues/41

File: telescope/selector.py [hunk 5]
```diff
@@ -87,25 +91,70 @@ def _parse_file_contents(self, selector_file_contents):
     selector_input_json = json.loads(selector_file_contents)
     self.validate_selector_input(selector_input_json)
 
-    metrics = []
     if selector_input_json['metric'] == 'all':
```
Comment (mtlynch) [inline telescope/selector.py]: I think we should just support metrics as a list and get rid of the `all` meta-metric.

File: telescope/selector.py [hunk 6]
```diff
@@ -87,25 +91,70 @@ def _parse_file_contents(self, selector_file_contents):
...
-      metrics.append(selector_input_json['metric'])
+      selector_input_json['metric'] = self.supported_metrics.keys()
+
+    selector_dicts = self._parse_input_for_selectors(selector_input_json)
+    selectors = self._create_selectors_from_dict(selector_dicts)
+    
+    return selectors
 
+  def _parse_input_for_selectors(self, selector_dict):
+    """ Parse the selector JSON diction and return a list of dictionaries
```
Comment (mtlynch) [inline telescope/selector.py]: Is `diction` supposed to be `dictionary`?

File: telescope/selector.py [hunk 7]
```diff
@@ -87,25 +91,70 @@ def _parse_file_contents(self, selector_file_contents):
...
+      flattened for each combination.
+      
+      Args:
+        selector_dict (dict): Dictionary parsed from a valid
```
Comment (mtlynch) [inline telescope/selector.py]: Can we call this `selector_json`?

File: telescope/selector.py [hunk 8]
```diff
@@ -87,25 +91,70 @@ def _parse_file_contents(self, selector_file_contents):
...
+      
+      Returns:
+        list: List of dictaries representing the possible combinations of
+          the selector query.
+      
+    """
     selectors = []
+    has_not_recursed = True
+    
+    for iterable_field_key in self.iterable_field_keys:
```
Comment (mtlynch) [inline telescope/selector.py]: I think it's possible to do this with less complexity using `itertools`. Could we do something like this:

```
start_times = selector_json['start_time']
client_providers = selector_json['client_provider']
sites = selector_json['site']
metrics = selector_json['metric']

for start_time, client_provider, site, metric in itertools.product(start_times, client_providers, sites, metrics):
  selector = Selector()
  selector.start_time = start_time
  ...
```

That way we don't have to deal with making lots of copies of the JSON or passing around these sort of "not quite real `Selector` objects yet" selector `dicts`.

**Comment by mtlynch** [comment]
It won't let me comment on it, but there's a .DS_Store file that got added in. Can you take it out and add *.DS_Store to the root .gitignore?

Can you also add unit tests for the new selector format?

File: telescope/selector.py (hunk 5 above)
Comment (collina) [inline telescope/selector.py]: Happily. That fixes our changing-meaning-of-all problem.

File: main.py (hunk 3 above)
Comment (collina) [inline main.py]: Two hours of my evening sunk into a minor git issue caused by committing .DS_Store.  I cleaned everything up but this showed. Not completely sure.

**Comment by collina** [comment]
Did it get readded? Ugh.

File: telescope/selector.py (hunk 8 above)
Comment (collina) [inline telescope/selector.py]: Didn't know itertools could do that. Saves a lot of trouble.

File: documentation/examples/interconnection_study_example.json (hunk 1 above)
Comment (collina) [inline documentation/examples/interconnection_study_example.json]: In the selector, but internally (main.py and onward) I think we should keep them singular.

**Comment by mtlynch** [comment]
LGTM

**Pull request merged**